discovery/utils/dataframe_matcher.py

```diff
@@ -72,3 +85,93 @@ def _match_columns(self, reference_column_name, subject_column_name):
 
         # Produce a similarity percentage as an average of all checks
         return round(sum(percentages) / len(percentages), 2)
```

Comment:
Long term this is probably better as a dictionary of match types and each of their confidences